packages/babel-plugin/src/styled/__tests__/tagged-template-expression.test.ts

```diff
@@ -819,6 +918,120 @@ describe('styled tagged template expression', () => {
+
+  it('should transform variable within a template literal within an interpolation function', () => {
+    const actual = transform(
+      `
+      import { styled } from '@compiled/react';
+
+      const color = 'red';
+      const color2 = 'blue';
+
+      const ListItem = styled.div<{ ruleEnabled: boolean }>\`
+        color: \${({ ruleEnabled }) =>
+          (ruleEnabled ? color : \`var(--my-variable, \${color2})\`)};
+      \`;
+    `,
+      enableTypescript
+    );
+
+    expect(actual).toMatchInlineSnapshot(`
+      "const _2 = "._syazj09m{color:var(--my-variable,blue)}";
+      const _ = "._syaz5scu{color:red}";
+      const color = "red";
+      const color2 = "blue";
+      const ListItem = forwardRef(
+        ({ as: C = "div", style: __cmpls, ...__cmplp }, __cmplr) => {
+          const { ruleEnabled, ...__cmpldp } = __cmplp;
+          return (
+            <CC>
+              <CS>{[_, _2]}</CS>
+              <C
+                {...__cmpldp}
+                style={__cmpls}
+                ref={__cmplr}
+                className={ax([
+                  "",
+                  __cmplp.ruleEnabled ? "_syaz5scu" : "_syazj09m",
+                  __cmplp.className,
+                ])}
+              />
+            </CC>
+          );
+        }
+      );
+      "
+    `);
+  });
+
+  it('should transform variables within nested template literals that are all in an interpolation function', () => {
+    // this aims to represent the following use case:
+    //     background-color: ${({ isActive }) =>
+    //         `${
+    //             isActive
+    //                 ? `${token('some.token', color)}`
+    //                 : `${token('some.other.token', color2)}`
+    //         }`
+    //     };
+    //
+    // after token(...) has been processed by the babel plugin in
+    // @atlaskit/tokens (as of @atlaskit/tokens v1.0.0)
+    const actual = transform(
+      `
+      import { styled } from '@compiled/react';
+
+      const color = 'red';
+      const color2 = 'blue';
+
+      const ListItem = styled.div<{ isActive: boolean }>\`
+        background-color: \${({ isActive }) =>
+          \`\${
+            isActive
+              ? \`\${\`var(--my-variable, color)\`}\`
+              : \`\${\`var(--my-other-variable, color2)\`}\`
+          }\`
+        };
+      \`;
+    `,
+      enableTypescript
+    );
+
+    expect(actual).toMatchInlineSnapshot(`
+      "const _ = "._bfhk17td{background-color:var(--_hcoy15)}";
+      const color = "red";
+      const color2 = "blue";
+      const ListItem = forwardRef(
+        ({ as: C = "div", style: __cmpls, ...__cmplp }, __cmplr) => {
+          const { isActive, ...__cmpldp } = __cmplp;
+          return (
+            <CC>
+              <CS>{[_]}</CS>
+              <C
+                {...__cmpldp}
+                style={{
+                  ...__cmpls,
+                  "--_hcoy15": ix(
+                    \`\${
+                      __cmplp.isActive
```

Comment:
Just wondering if `var(--my-variable, color)` and `var(--my-other-variable, color2)` are supposed to be part of style attribute?

Reply:
Ah good question! To summarise and expand on our Zoom discussion from yesterday, yep that is correct - my understanding is that since we have a ternary expression here, it makes sense to not try to evaluate it to a regular string at build time. This is because we don't really know what the value of `background-color` will be at build time, which in turn is because we might not know whether `isActive` is true or false before runtime.

Slight tangent: I suppose we *could* evaluate just `color` and `color2` at build time so that instead of having `var(--my-variable, ${color})` in the babel plugin output, we would have `var(--my-variable, red)`. (We still keep the ternary expression in the babel plugin output of course.) However from what I can tell, both versions generate the exact same HTML `style="--_hudwc7:var(--my-variable, red);"`.

Thank you for bringing this up - looking at this again, I see there's actually a typo in this test case that I need to fix 😆 `var(--my-variable, color)` should be `var(--my-variable, ${color})`